### ISSUE TITLE
System.Reflection.Metadata.AssemblyExtensionsTest.ApplyUpdateInvalidParameters failing intermittently 

### ISSUE THREAD

**Issue by hoyosjs** [dotnet/runtime#48523]
![image](https://user-images.githubusercontent.com/19413848/108555749-c16bb400-72aa-11eb-876f-54911eee8de0.png)

On CI, this is a test that's 90% reliable for rolling builds according to AzDO. 

cc: @mikem8361 

PS: Jared, do you know why the [runfo query](https://runfo.azurewebsites.net/search/tests/?bq=definition%3Aruntime%20%20started%3A~7&tq=name%3A%22system.reflection.metadata.assemblyextensionstest.applyupdateinvalidparameters%22) only shows two hits while [AzDO analytics](https://dev.azure.com/dnceng/public/_test/analytics?definitionId=686&contextType=build) shows 8 hits?

**Comment by hoyosjs** [comment]
@Anipik

**Comment by mikem8361** [comment]
My next PR will fix the failures on Windows arm/arm64.

**Comment by VincentBu** [comment]
Same test failed in [runtime 20210219.53](https://dev.azure.com/dnceng/public/_build/results?buildId=1003415&view=ms.vss-test-web.build-test-results-tab&runId=31346398&resultId=150220&paneView=debug)

Faield test:
```
net6.0-windows-Release-arm64-CoreCLR_release-Windows.10.Arm64.Open
 -System.Reflection.Metadata.AssemblyExtensionsTest.ApplyUpdateInvalidParameters
```

**Error message:**
```
Assert.Throws() Failure
Expected: typeof(System.InvalidOperationException)
Actual: typeof(System.NotImplementedException): The method or operation is not implemented.
---- System.NotImplementedException : The method or operation is not implemented.

Stack trace
   at System.Reflection.Metadata.AssemblyExtensions.ApplyUpdate(Assembly assembly, ReadOnlySpan`1 metadataDelta, ReadOnlySpan`1 ilDelta, ReadOnlySpan`1 pdbDelta) in /_/src/coreclr/System.Private.CoreLib/src/System/Reflection/Metadata/AssemblyExtensions.cs:line 80
   at System.Reflection.Metadata.AssemblyExtensionsTest.<>c__DisplayClass1_0.<ApplyUpdateInvalidParameters>b__2() in /_/src/libraries/System.Runtime.Loader/tests/AssemblyExtensionsTest.cs:line 34
----- Inner Stack Trace -----
   at System.Reflection.Metadata.AssemblyExtensions.ApplyUpdate(Assembly assembly, ReadOnlySpan`1 metadataDelta, ReadOnlySpan`1 ilDelta, ReadOnlySpan`1 pdbDelta) in /_/src/coreclr/System.Private.CoreLib/src/System/Reflection/Metadata/AssemblyExtensions.cs:line 80
   at System.Reflection.Metadata.AssemblyExtensionsTest.<>c__DisplayClass1_0.<ApplyUpdateInvalidParameters>b__2() in /_/src/libraries/System.Runtime.Loader/tests/AssemblyExtensionsTest.cs:line 34
```

**Comment by mikem8361** [comment]
Fixed with PR #48497
